tests/test_virtio_blk.py

```diff
@@ -16,9 +16,43 @@ def test_loading_virtio_mmio(helpers: conftest.Helpers) -> None:
         assert res.stdout.find("virtio_mmio") >= 0
 
 
+def test_userfaultfd_completes(helpers: conftest.Helpers) -> None:
+    with helpers.spawn_qemu(helpers.notos_image()) as vm:
+        vmsh = helpers.spawn_vmsh_command(
+            ["guest_userfaultfd", str(vm.pid)],
+            cargo_executable="test_ioctls",
+            stdout=subprocess.PIPE,
+        )
+
+        try:
+            helpers.vmsh_print_stdout_until(vmsh, "pause\n")
+            vm.wait_for_ssh()
+            print("ssh available")
+
+            res = vm.ssh_cmd(
+                [
+                    "devmem2",
+                    "0xd0000000",
+                    "ww",
+                    "0x1337",
+                ]
+            )
+            print("stdout:\n", res.stdout)
+            print("stderr:\n", res.stderr)
+
+        finally:
+            # we cannot kill sudo, but we can stop vmsh as it drops privileges to our user
+            subprocess.run(["pkill", "--parent", str(vmsh.pid)])
+            vmsh.wait()
```

Comment:
I am going to move this to a helper function...

Reply:
ok